### PR TITLE
Load obj cleanup

### PR DESCRIPTION
**Pull request by AntonPuko**

### CODE REVIEW

File: src/dao/BaseDAO.js [hunk 1]
```diff
@@ -2,26 +2,23 @@
...
-import { createFilter } from '../filters';
 
 const ID_REG_EXP = /\bid\b/;
 
 class BaseDAO<TEntity, TEntityMutator> {
   static _organizationID: ?string = null;
   __config: DAOConfig<TEntity, TEntityMutator>;
-  _entityLoaderByID: Map<EntityID, LoadObject<TEntity>> = new Map();
-  _entityIDsLoaderByQuery: Map<string, LoadObject<Array<EntityID>>> = new Map();
+  _entityLoaderByID: Map<string, LoadObject<TEntity>> = new Map();
```
Comment (jlkalberer) [inline src/dao/BaseDAO.js]: Even though `EntityID` is just a string, I still want to use it as a type.

It makes things more readable and it's easier to understand that you're working with an ID

File: src/dao/BaseDAO.js [hunk 2]
```diff
@@ -58,8 +55,9 @@ class BaseDAO<TEntity, TEntityMutator> {
   __setupHandler(
     handler: oHandler<TEntity>,
     queryOptions?: QueryOptions = {},
-    shouldSelectExpand: boolean = true
+    shouldSelectExpand: boolean = true,
   ): oHandler<TEntity> {
+    /* eslint-disable no-param-reassign */
```
Comment (jlkalberer) [inline src/dao/BaseDAO.js]: Do you still need this?

I actually think that `handler` isn't immutable so we probably don't need to reassign.

File: src/dao/DAO.js [hunk 3]
```diff
@@ -44,42 +26,43 @@ class DAO<TEntity: { id: EntityID }, TEntityMutator> extends BaseDAO<
...
       .then(() => {
         this._entityLoaderByID.delete(id);
         this._clearQueryCaches();
         this._emitChanges();
       })
-      .catch((error) => this._updateCacheForError(id, error));
+      .catch((error: Error): void => this._updateCacheForError(id, error));
   }
 
   count(queryOptions?: QueryOptions): LoadObject<number> {
```
Comment (jlkalberer) [inline src/dao/DAO.js]: Can you test this or have you tested it?

Comment (AntonPuko) [reply src/dao/DAO.js]: ok, fixed

Comment (jlkalberer) [reply src/dao/DAO.js]: Does the query actually return entities? We should figure out a way to just return the count if it does.

Comment (AntonPuko) [reply src/dao/DAO.js]: Response from the server looks like {data:[], inlinecount: 123}, data is blank array

**Pull request merged**